In [8]:
import csv 

with open('DataSets/Question2BupaOriginal.data','r') as data_file:
    lines = data_file.readlines()

with open('DataSets/CsvBupa.csv','w', newline='') as csv_file:
    csv_writer = csv.writer(csv_file)
    for line in lines:
        # remove any whitespace from the beginning and end of the line
        line = line.strip()
        
        line = line.strip('"')
        # split the line into fields using whitespace as the delimiter
        fields = line.split()

        # write the fields to the csv file
        csv_writer.writerow(fields)

In [9]:


with open('DataSets/CsvBupa.csv','r') as file:
    reader = csv.reader(file)
    data = list(reader)

columns = [list(x[0].split(',')) for x in data]

rows = list(map(list, zip(*columns)))

rows = zip(*rows)
with open('DataSets/CsvBupaFormatted.csv','w', newline='') as formatted_csv:
    writer = csv.writer(formatted_csv)
    writer.writerows(rows)

Now that data is sorted into an easier to manage format, we will determine the cutoff point to dichotomise the class variable, we will also remove the 7th column as it is not needed, and normalize the data after it has been dichotomised.

In [40]:

import pandas as pd
from sklearn import preprocessing
data_frame = pd.read_csv('DataSets/CsvBupaFormatted.csv',header = None)
data_frame = data_frame.drop(6, axis=1)
cutoff_point = data_frame[5].mean()
print(cutoff_point)
data_frame[5] = data_frame[5].apply(lambda x: 0 if x < cutoff_point else 1)
explanatory = data_frame[[1,2,3,4]]
scaler = preprocessing.StandardScaler()
explanatory = scaler.fit_transform(explanatory)
normalized_df = pd.DataFrame(explanatory,columns=[1,2,3,4])
normalized_df[5] = data_frame[5]
print(normalized_df)


3.455072463768116
            1         2         3         4  5
0    1.207923  0.749035  0.234482 -0.185829  0
1   -0.320373  1.467573  0.732000 -0.389922  0
2   -0.866193  0.133145 -0.860056  0.400941  0
3    0.443775  0.184469 -0.064028 -0.058270  0
4    0.007119 -0.944662  0.333986 -0.721575  0
..        ...       ...       ...       ... ..
340  0.280029 -0.226124 -0.064028  0.069288  1
341 -0.047463  1.159628  1.826538  4.202186  1
342  0.389193  1.262276  1.030510  1.293850  1
343 -0.102045 -0.174800  0.134979 -0.619528  1
344  1.589997  1.364925  2.025545  0.681569  1

[345 rows x 5 columns]


Since the mean is 3.455 we will set the cutoff point as => 3.5.
So if a value is less then that it will be set to 0 and if it is greater it will be set to one.

            1         2         3         4  5
0    1.207923  0.749035  0.234482 -0.185829  0
1   -0.320373  1.467573  0.732000 -0.389922  0
2   -0.866193  0.133145 -0.860056  0.400941  0
3    0.443775  0.184469 -0.064028 -0.058270  0
4    0.007119 -0.944662  0.333986 -0.721575  0
..        ...       ...       ...       ... ..
340  0.280029 -0.226124 -0.064028  0.069288  1
341 -0.047463  1.159628  1.826538  4.202186  1
342  0.389193  1.262276  1.030510  1.293850  1
343 -0.102045 -0.174800  0.134979 -0.619528  1
344  1.589997  1.364925  2.025545  0.681569  1

[345 rows x 5 columns]
